docs: add designer handboek "breaking changes" (#1151)
nl-design-system/documentatie: docs/handboek/designer/breaking-changes.md

---

> title: Breaking changes

Reviewer: Eensch

### Minor

Het ophogen van het minor deel van het versienummer gebeurt als er veranderingen zijn doorgevoerd die _wel_ achterwaarts compatibel zijn. Denk daarbij aan nieuwe features, of nog specifieker nieuwe design tokens. Bij het verhogen van het minor deel van het versienummer blijft het major deel hetzelfde en springt alleen het patch deel terug naar 0.

### Patch

Het ophogen van het patch deel van het versienummer gebeurt in de regel als er bugs zijn opgelost. Deze veranderingen zijn altijd achterwaarts compatibel. Bij het verhogen van het patch deel van het versienummer blijven het major en minor deel hetzelfde.

Een versienummer alleen zegt niet zo veel. Door de versienummers met elkaar te vergelijken kun je als afnemer van een package zien wat de impact is van de veranderingen op code die gebruik maakt van het package.

## Hoe bepaal je het type van je verandering?

### Breaking changes

Breaking changes in design tokens zijn veranderingen die voor afnemers van je package negatieve gevolgen kunnen hebben. Het zijn veranderingen die niet achterwaarts compatibel zijn waardoor bestaande code niet meer werkt zoals voorheen.

> Het resultaat hiervan is op de een of andere manier kapot. Soms is dit heel duidelijk, bijvoorbeeld spacing die helemaal is verdwenen. Soms is het subtieler en misschien niet meteen zichtbaar, een hover kleur die is aangepast.

Author: Ik laat het voor nu weg. Misschien dat we nog een keer een beter voorbeeld kunnen bedenken.

Breaking changes zijn zoals [hierboven uitgelegd](#major), veranderingen die ervoor zorgen dat het **major** deel van het versienummer opgehoogd moet worden.

> #### Voorbeelden van breaking changes in design tokens

Author: Ja… zouden we voorbeelden van waarde veranderingen voor zowel major, minor, als patch versie bumps kunnen bedenken?

> #### Voorbeelden van breaking changes in design tokens

Reviewer 2: Voorbeelden zijn toegevoegd, maar nog niet zo uitgebreid als een tabel of lijst. @Rozerinay Zou jij hier een ticket voor kunnen aanmaken, en dan aan de hand van de changes die jij designers ziet maken, een tabel of lijst maken? Dan kan deze PR door

> #### Voorbeelden van breaking changes in design tokens

Reviewer 3: Ik heb het op mijn lijstje gezet!

> De eenvoudigste soort breaking change van een design token ontstaat als je een design token verwijdert. Gebruikers die dit verwijderde design token in hun code gebruiken zien bij het gebruik van een nieuwe versie dat hun code niet meer werkt zoals voorheen.

Author: In plaats van de kop die er nu staat of bedoel je eenzelfde of nog dieper niveau?

**Voorbeeld 1**: Het verwijderen van de design token `example.button.icon.space` zou ervoor zorgen dat de ruimte tussen het icoon en de tekst van een button komt te vervallen. De button gaat door deze verandering kapot.

> Een ander soort breaking change die hier op lijkt is als je de naam van een design token verandert. Dit lijkt misschien een minder grote verandering dan het verwijderen van een design token maar in feite komt het neer op:

Author: Het zijn maar voorbeelden, ik denk dat kopjes een eindige lijst zou suggereren en dat is het niet per se.

1. Het verwijderen van een design token.
1. Het toevoegen van een nieuw design token.

**Voorbeeld 2**: Het hernoemen van de design token `example.button.icon.space` naar `example.button.column-gap` zorgt ervoor dat de design token `example.button.icon.space` niet meer bestaat. Net als in het vorige voorbeeld gaat de button door deze verandering kapot als een afnemer van het package gebruik maakt van de verwijderde design token.

**Voorbeeld 3**: Het aanpassen van zelfs maar 1 karakter in de naam van een design token, bijvoorbeeld het corrigeren van een typfout `colour` naar `color`, kan al een breaking change zijn.

Breaking changes zijn, ondanks dat hun naam het doet vermoeden, niet per se erg. Het is alleen heel belangrijk om er duidelijk over te communiceren. Daarover later meer.

### Nieuwe features

> Nieuwe features, voor designers hoofdzakelijk nieuwe design tokens, zijn veranderingen die ervoor zorgen dat het **minor** deel van het versienummer opgehoogd moet worden.

Author: Ik denk dat ze beide belangrijk zijn. Ik noem ze beide als voorbeeld.

**Voorbeeld**: Je maakt een nieuw design token `example.button.box-shadow` aan dat gebruikt kan worden een button een drop shadow te geven.

### Bug fixes

> In Tokens Studio kun je aan design tokens beschrijvingen toevoegen. Als je in de beschrijving van een design token een spelfout corrigeert heeft dat geen gevolgen voor afnemers van de design token. Deze veranderingen lijken nog het meest op een bug fix maar zullen waarschijnlijk in de praktijk weinig voorkomen.

Author: Ik ga vandaag met Robbert praten want blijkbaar is Tokens Studio niet de enige plek waar waarheden over design tokens leven dus ik denk dat er nog wat meer gesleuteld moet worden.

## Breaking changes uitstellen

Om te voorkomen dat een wijziging meteen een breaking change is kun je ervoor kiezen om een nieuw design token naast een te verwijderen design te laten bestaan. Hierbij wordt het te verwijderen token eerst gemarkeerd als achterhaald (in het Engels ‘deprecated’) en niet meteen helemaal verwijderd.

**Voorbeeld**: De naam van de design token `example.button.icon.space` moet veranderen in `example.button.column-gap`.

1. Maak eerst de nieuwe design token `example.button.column-gap` aan.

> 1. Laat het oude design token `example.button.icon.space` verwijzen naar de waarde van de nieuwe design token: `{example.button.column-gap}`.

Reviewer 4: Wel lekker om elke zin te starten met een werkwoord?

'Verwijs de oude design token example.button.icon.space naar de waarde van de nieuwe design token: {example.button.column-gap}.'

> 1. Markeer de oude design token `example.button.icon.space` als ‘deprecated’ bijvoorbeeld door er een sticker bij te zetten of door de beschrijving aan te passen.

Reviewer 4: Ik denk dat we deze 3e stap moeten aanpassen als we eenmaal weten wat de juiste manier is om een depricated token te markeren.

'Depricated' in de beschrijving **en/of** Sticker.

Author: Kunnen we dat voor mensen die Tokens Studio voor design tokens gebruiken nu al vastleggen? Ik denk zelf dat [deprecated] in de description van de token _de_ manier zou moeten zijn. Het is de enige manier om [deprecated] ook in de JSON terecht te laten komen.

> Doordat je de oude token niet verwijdert maar markeert als ‘deprecated’ vertel je gebruikers die de design token al gebruiken dat ze voor nieuwe code de design token niet meer moeten gebruiken en dat ze bestaande code zullen moeten aanpassen. Hoe code moet worden aangepast geef je aan in een changelog. Meer daarover hieronder.

Reviewer 4: Lange zin. Voorzetje

Doordat je de oude token niet verwijdert maar markeert als ‘Deprecated’ informeer je afnemers van de design token dat ze deze design token niet meer moeten gebruiken. En dat ze bestaande code moeten gaan aanpassen

Author: Het is belangrijk om het onderscheid tussen nieuwe en bestaande code te hebben denk ik. Als je alleen maar zegt dat het niet meer gebruikt moet worden dan lijkt me dat mensen kunnen denken dat ze meteen nieuwe code moeten aanpassen. Deprecation zorgt er juist voor dat dat iets minder urgent is.

> Stel de huidige versie van je package is 1.x.x en je wilt op termijn een design token verwijderen. Afnemers van het package die nu versie 1.x.x gebruiken zien

Reviewer 4: Deze zin is nog niet af gok ik?

Author: Ja, weggehaald. Ik denk dat wij eerst moeten bepalen hoe de tijdslijn voor deprecation er ongeveer uit zou moeten zien voor ik er wat zinnigs over kan schrijven.

> ## Communiceer over je veranderingen

Reviewer 4: Hier kan misschien nog iets van een intro bij? Voorzet:

Jij weet wat je verandert hebt. De afnemer van de design tokens niet. Het is dus wel zo netjes om afnemers te informeren over veranderingen. We leggen je uit hoe je deze communicatie kan aanpakken.

Als je een [pull request hebt aangemaakt](/handboek/designer/stappenplan/#verstuur-aanpassingen-naar-github) geef je daar met een extra commit aan wat je precies veranderd hebt. De beschrijving die je hierbij opgeeft komt terecht in de changelog van het package.

> ### Situatie A: in het NL Design System “themes” repository

Reviewer 4: Misschien de stappen even introduceren?

**Situatie A: NL Design System “themes” repository**
Werk je vanuit de NL Design System “themes” repository? Volg dan onderstaande stappen.

1. Klik rechtsboven op ‘Add a file’ en kies voor ‘Create new file’.
1. Geef je bestand een willekeurige naam en de extensie `.md`, bijvoorbeeld `aap-noot-mies.md`.
1. Plak het gekopieerde sjabloon in het veld met ‘Enter file contents here’ en pas de inhoud aan.
1. Gebruik de juiste naam voor het package, namelijk `"@nl-design-system-unstable/{organisatie}-design-tokens"`.
1. Geef het type van je verandering(en) aan `major`, `minor`, of `patch`.
1. Geef een duidelijke beschrijving van je veranderingen.
1. Klik rechtsboven op ‘Commit changes…’.
1. Geef een commit message op, bijvoorbeeld ‘Changeset toegevoegd’.
1. Kies onderaan voor ‘Commit directly to the `{naam-van-je-branch}`’ branch.
1. Klik op ‘Commit changes’. Als je hier een foutmelding krijgt dat het bestand al bestaat kies je een andere willekeurige naam.

Hieronder een voorbeeld van de inhoud van het bestand met de beschrijving van een verandering met breaking changes voor de ‘voorbeeld’ organisatie.

```markdown
---
"@nl-design-system-unstable/voorbeeld-design-tokens": major
---

Deprecated design token example.button.icon.spacing verwijderd
```

> ### Situatie B: in je eigen repository

Reviewer 4: En dan hier ook?

**Situatie B: Eigen repository**
Werk je vanuit een eigen repository? Volg dan onderstaande stappen.

---
title: Versiebeheer voor design tokens
---

> ## Inleiding

Reviewer 2: Verwerkt!

Design tokens komen uiteindelijk terecht in een package. Een package is een pakketje van code dat gebruikt kan worden door ontwikkelaars. Denk daarbij aan CSS variabelen.

Deze packages worden geüpload naar een package registry. Een package registry is een database van alle packages die ooit zijn geüpload.

Alle packages in een registry hebben een versienummer. Als er een nieuwe versie van een package wordt geüpload wordt het versienummer van het package opgehoogd.

## Semantic versioning in een notendop

Bij NL Design System maken we voor het ophogen van versienummers gebruik van ‘semantic versioning’ ook wel bekend als semver. Binnen semver bestaat een versienummer uit drie delen: `major.minor.patch` (bijvoorbeeld `1.3.2`).

> 

Reviewer 2: Verwerkt!

### Minor

Het ophogen van het minor deel van het versienummer gebeurt als er veranderingen zijn doorgevoerd die _wel_ achterwaarts compatibel zijn. Denk daarbij aan nieuwe features, of nog specifieker nieuwe design tokens. Bij het verhogen van het minor deel van het versienummer blijft het major deel hetzelfde en springt alleen het patch deel terug naar 0.

### Patch

Het ophogen van het patch deel van het versienummer gebeurt in de regel als er bugs zijn opgelost. Deze veranderingen zijn altijd achterwaarts compatibel. Bij het verhogen van het patch deel van het versienummer blijven het major en minor deel hetzelfde.

### Waarom zijn versienummers belangrijk?

Een versienummer alleen zegt niet zo veel. Door de versienummers met elkaar te vergelijken kun je als afnemer van een package zien wat de impact is van de veranderingen op code die gebruik maakt van het package.

## Hoe bepaal je het type van je verandering?

### Breaking changes

Breaking changes in design tokens zijn veranderingen die voor afnemers van je package negatieve gevolgen kunnen hebben. Het zijn veranderingen die niet achterwaarts compatibel zijn waardoor bestaande code niet meer werkt zoals voorheen.

> Er gaat op de een of andere manier iets kapot. Soms is dit heel duidelijk, bijvoorbeeld spacing die helemaal is verdwenen. Soms is het subtieler en misschien niet meteen zichtbaar.

Reviewer 2: Is zoals bijvoorbeeld niet dubbel? https://taaladvies.net/zoals-bijvoorbeeld-dubbelop/#:~:text=Toelichting,bijvoorbeeld%2C%20maar%20dat%20is%20dubbelop.

Breaking changes zijn zoals [hierboven uitgelegd](#major), veranderingen die ervoor zorgen dat het **major** deel van het versienummer opgehoogd moet worden.

#### Voorbeelden van breaking changes in design tokens

De eenvoudigste soort breaking change van een design token ontstaat als je een design token verwijdert. Afnemers die dit verwijderde design token in hun code gebruiken zien bij het gebruik van een nieuwe versie dat hun code niet meer werkt zoals voorheen.

> **Voorbeeld 1**: Het verwijderen van de design token `example.button.icon.space` zou ervoor zorgen dat de ruimte tussen het icoon en de tekst van een button komt te vervallen. De button gaat door deze verandering kapot.

Reviewer 2: Verwerkt!

Een ander soort breaking change die hier op lijkt is als je de naam van een design token verandert. Dit lijkt misschien een minder grote verandering dan het verwijderen van een design token maar in feite komt het neer op:

1. Het verwijderen van een design token.

> 1. Het toevoegen van een nieuw design token.

Reviewer 3: nieuwe* design token.

> **Voorbeeld 2**: Het hernoemen van de design token `example.button.icon.space` naar `example.button.column-gap` zorgt ervoor dat de design token `example.button.icon.space` niet meer bestaat. Net als in het vorige voorbeeld gaat de button door deze verandering kapot als een afnemer van het package gebruik maakt van de verwijderde design token.

Reviewer 3: Als we uitgaan van **knop** i.p.v. **button**, dan moeten we dat woord hier ook nog aanpassen.

> **Voorbeeld 2**: Het hernoemen van de design token `example.button.icon.space` naar `example.button.column-gap` zorgt ervoor dat de design token `example.button.icon.space` niet meer bestaat. Net als in het vorige voorbeeld gaat de button door deze verandering kapot als een afnemer van het package gebruik maakt van de verwijderde design token.

Reviewer 3: Zou van de laatste zin maken:

> Net als in het eerste voorbeeld kan dit ervoor zorgen dat de knop niet meer naar behoren functioneert voor afnemers die de verwijderde design token gebruiken.

> **Voorbeeld 3**: Het aanpassen van zelfs maar 1 karakter in de naam van een design token, bijvoorbeeld het corrigeren van een typfout `colour` naar `color`, kan al een breaking change zijn.

Reviewer 3: Zou hiervan maken:

> Zelfs het aanpassen van slechts één karakter in de naam van een design token, bijvoorbeeld het corrigeren van een typfout van `colour` naar `color`, kan al als een breaking change worden beschouwd.

Author: Ik heb er 

> Zelfs het aanpassen van slechts één karakter in de naam van een design token, bijvoorbeeld het corrigeren van een typfout `colour` naar `color`, is een breaking change.

van gemaakt omdat het een breaking change is.

> Breaking changes zijn, ondanks dat hun naam het doet vermoeden, niet per se erg. Het is alleen heel belangrijk om er duidelijk over te communiceren. Daarover later meer.

Reviewer 3: Zou hiervan maken:

> Breaking changes zijn, ondanks hun naam, niet per se problematisch. Het is echter van groot belang om hier duidelijk over te communiceren. Daarover meer informatie later.

### Nieuwe features

> Nieuwe features, bijvoorbeeld nieuwe design tokens of een nieuwe property ‘dismissable‘, zijn veranderingen die ervoor zorgen dat het **minor** deel van het versienummer opgehoogd moet worden.

Reviewer 3: Zou hiervan maken:

> Nieuwe features, zoals nieuwe design tokens of een nieuwe eigenschap ‘dismissable’, zijn veranderingen die ervoor zorgen dat het **minor** deel van het versienummer moet worden opgehoogd.

**Voorbeeld**: Je maakt een nieuw design token `example.button.box-shadow` aan dat gebruikt kan worden een button een drop shadow te geven.

### Bug fixes

> In Tokens Studio kun je aan design tokens beschrijvingen toevoegen. Als je in de beschrijving van een design token een spelfout corrigeert heeft dat geen gevolgen voor afnemers van de design token. Deze veranderingen lijken nog het meest op een bug fix maar zullen waarschijnlijk in de praktijk weinig voorkomen.

Reviewer 3: Zou hiervan maken:

> In Tokens Studio kun je beschrijvingen aan design tokens toevoegen. Het corrigeren van een spelfout in de beschrijving van een design token heeft geen gevolgen voor de afnemers. Deze veranderingen lijken het meest op een bug fix, maar zullen in de praktijk waarschijnlijk weinig voorkomen.

## Breaking changes uitstellen

> Om te voorkomen dat een wijziging meteen een breaking change is kun je ervoor kiezen om een nieuw design token naast een te verwijderen design te laten bestaan. Hierbij wordt het te verwijderen token eerst gemarkeerd als ‘deprecated’ (in het Nederlands ‘achterhaald’) en niet direct verwijderd.

Reviewer 4: ```suggestion
Om te voorkomen dat een wijziging meteen een breaking change is kun je ervoor kiezen om een nieuwe design token naast een te verwijderen design token te laten bestaan. Hierbij wordt de te verwijderen design token eerst gemarkeerd als ‘deprecated’ (in het Nederlands ‘achterhaald’) en niet direct verwijderd.
```

> **Voorbeeld**: De naam van de design token `example.button.icon.space` moet veranderen in `example.button.column-gap`.

Reviewer 2: Ik zie het inderdaad door elkaar gebruikt worden in de hele repo. Laat het me weten of we dat willen rechttrekken, dan maak ik daar een aparte PR voor aan

> **Voorbeeld**: De naam van de design token `example.button.icon.space` moet veranderen in `example.button.column-gap`.

Author: We zouden ook `{prefix}` kunnen gebruiken dan dekt het zowel de lading voor `nl`, als community prefixes.

> **Voorbeeld**: De naam van de design token `example.button.icon.space` moet veranderen in `example.button.column-gap`.

Reviewer 2: Ik heb er deze ticket voor aangemaakt: https://github.com/nl-design-system/documentatie/issues/2159 zodat deze conversation en PR ready is

1. Maak eerst de nieuwe design token `example.button.column-gap` aan.
1. Verander de waarde van de oude design token `example.button.icon.space` naar de naam van de nieuwe design token: `{example.button.column-gap}`.
1. Markeer de oude design token `example.button.icon.space` als ‘deprecated’ door `[deprecated]` aan de beschrijving van de token toe te voegen.

> Doordat je de oude token niet verwijdert maar markeert als ‘deprecated’ informeer je afnemers dat ze voor nieuwe code de design token niet meer moeten gebruiken. Ze weten ook dat te op termijn bestaande code zullen moeten aanpassen. Hoe code moet worden aangepast geef je aan in een changelog. Meer daarover hieronder.

Reviewer 3: Zou hiervan maken:

> Door de oude token niet te verwijderen maar te markeren als ‘deprecated’, informeer je afnemers dat ze deze design token niet meer moeten gebruiken. Ze weten ook dat ze op termijn bestaande code zullen moeten aanpassen. Hoe de code moet worden aangepast, geef je aan in een changelog. Meer daarover hieronder.

Reviewer 4: Ik zou deze comment van Roos graag nog doorvoeren. Dat lost ook de schrijffout 'Ze weten ook dat te op...' op.

1. Geef een duidelijke beschrijving van je veranderingen.
1. Klik rechtsboven op ‘Commit changes…’.
1. Geef een commit message op, bijvoorbeeld ‘Changeset toegevoegd’.
1. Kies onderaan voor ‘Commit directly to the `{naam-van-je-branch}`’ branch.
1. Klik op ‘Commit changes’. Als je een foutmelding krijgt dat het bestand al bestaat, kies dan een andere willekeurige naam.

Hieronder een voorbeeld van de inhoud van het bestand met de beschrijving van een verandering met breaking changes voor de ‘voorbeeld’ organisatie.

```markdown
---
"@nl-design-system-unstable/voorbeeld-design-tokens": major
---

Deprecated design token example.button.icon.spacing verwijderd
```

### Situatie B: in je eigen repository

Werk je vanuit een eigen repository? Volg dan onderstaande stappen.

Ga naar de lijst met pull requests en zoek je eigen pull request op.

<!-- screenshot pull requests -->

1. Klik boven in de pagina op de lichtblauwe naam van de branch die je hebt opgegeven.
1. Navigeer op de volgende pagina naar de folder `.changesets`.
1. Kopieer het sjabloon uit het `README.md` bestand op deze pagina naar je klembord.
1. Klik rechtsboven op ‘Add a file’ en kies voor ‘Create new file’.

> 1. Geef je bestand een willekeurige naam en de extensie `.md`, bijvoorbeeld `aap-noot-mies.md`.

Reviewer 2: Bij het automatisch genereren gebruikt changeset automatisch drie woorden, maar de naamgeving is volledig vrij en kan alles zijn. De file word vervolgens ook weer automatisch verwijderd zodra het is verwerkt in versienummer en changelog. Dus in dit geval zou ik zeggen; niet nodig

> 1. Geef je bestand een willekeurige naam en de extensie `.md`, bijvoorbeeld `aap-noot-mies.md`.

Author: Ik zou aanraden om het script te gebruiken maar als je dat niet doet voor de zekerheid + consistentie wel `{woord}-{woord}-{woord}.md` aanraden eigenlijk.

1. Plak het gekopieerde sjabloon in het veld met ‘Enter file contents here’ en pas de inhoud aan.
1. Geef het type van je verandering(en) aan: `major`, `minor`, of `patch`.
1. Geef een duidelijke beschrijving van je veranderingen.
1. Klik rechtsboven op ‘Commit changes…’.
1. Geef een commit message op, bijvoorbeeld ‘Changeset toegevoegd’.
1. Kies onderaan voor ‘Commit directly to the `{naam-van-je-branch}`’ branch.
1. Klik op ‘Commit changes’. Als je een foutmelding krijgt dat het bestand al bestaat, kies dan een andere willekeurige naam.

## Hoe verder?

> Nadat je pull request met veranderingen is goedgekeurd, wordt er op GitHub een nieuwe pull request aangemaakt die ervoor zorgt dat je veranderingen worden gepubliceerd naar het registry.

Reviewer 4: Mist hier nog de stap dat je een developer een seintje moet geven?

Reviewer 2: Gebaseerd op wat we hebben in "docs/semver" voor developer, wat vind je hiervan:

Na het mergen van je goedgekeurde pull request wordt er automatisch een nieuwe pull request aangemaakt om de versie te publiceren. Je kunt een developer vragen deze direct te mergen of te wachten tot er meer wijzigingen zijn. De publicatie-pull request zal na het mergen van verdere wijzigen steeds automatisch worden bijgewerkt. Zodra de publicatie-pull request gemerged wordt, worden de pakketten automatisch gepubliceerd op npm.

Author: Ik weet niet of dat hier hoort…

Je kunt er ook heel bewust voor kiezen om de PR open te laten omdat je nu al weet dat er meer changes aan zitten te komen. Maar om dat hier nu allemaal op te gaan schrijven schiet misschien het doel voorbij. Thoughts?

Reviewer 2: Ik wil wel "iets" in Hoe verder zetten, omdat het nu eindigt op, je hebt een pull request aangemaakt en dan ben je klaar. Terwijl die pull request dan forever open blijft staan, niemand heeft er ownership over. Gesprekje in thread https://codefornl.slack.com/archives/C02CWGL2M18/p1742545344220869

Reviewer 2: Voor de overige feedback heb ik deze tickets aangemaakt: https://github.com/nl-design-system/documentatie/issues/2190 en https://github.com/nl-design-system/documentatie/issues/2192

> ---

Reviewer 4: Algemeen:
- We schrijven altijd punten aan het eind van een opsomming. Dit heb ik overal doorgevoerd.
- Ik gok dat je niet voor niks voor een ordered opsomming kiest. Dus ik telde daarom de cijfers door.
- Als we een woord / term introduceren, of een woord benoemen uit een interface (Figma, Token Studio, Github, etc.) dan gebruiken we 'enkele haakjes'. Dit heb ik overal doorgevoerd.
- Je schrijft nu soms 'het design token'. Ik zeg zelf altijd 'de design token'. Zullen we even in de kernteam channel vragen wat we gaan aanhouden vanaf nu?

title: Breaking changes
---

> ## Breaking changes

Reviewer 4: Deze heading (h2) is nu gelijk aan de title (h1) das jammer ;)

Deze sectie is naar mijn idee een soort samenvatting van dit document? Dit hele stuk zou wat mij betreft dan ook als samenvatting (heading: Samenvattend) onderaan mogen staan. Als we hem bovenaan houden zou de heading 'Breaking changes in het kort' misschien toepasselijk zijn?

Reviewer 4: Daarnaast komt dit Breaking changes een beetje uit de lucht vallen. Vragen die ik direct had:

- Waar is dit voor?
- Wat zijn breaking changes (wordt verderop pas toegelicht)?
- Is dit alleen voor design tokens?
- Klinkt technisch. Waarom moet ik hier iets mee als designer? --> Ik weet het inmiddels maar niet iedereen :)
- Wanneer gaat dit voor mij als designer spelen? Ik ben nu nog e.e.a. aan het uitproberen. Wanneer wordt het zo serieus?

Reviewer 4: Verderop lees ik over 'minor' en 'patch'. Hoe verhoud dit zich tot breaking changes?

> 1. Je hebt een design token verwijderd

Reviewer 4: We schrijven altijd punten aan het eind van een opsomming. Ik zal dat verderop overal doorvoeren.
```suggestion
1. Je hebt een design token verwijderd.
```

> 1. Je hebt een design token van naam veranderd. Dit is hetzelfde als het verwijderen van een design token gevolgd door het toevoegen van een nieuw design token. Dat maakt het een breaking change

Reviewer 4: ```suggestion
1. Je hebt een design token van naam veranderd. Dit is hetzelfde als het verwijderen van een design token gevolgd door het toevoegen van een nieuw design token. Dat maakt het een breaking change.
```

> 1. Je hebt een spelfout in een design token gecorrigeerd. Dit lijkt omdat het zo'n kleine wijziging is misschien een patch versie ophoging. Ook dit is feitelijk hetzelfde als het verwijderen van een design token gevolgd door het toevoegen van een nieuw design token. Dat maakt ook dit een breaking change

Reviewer 4: ```suggestion
1. Je hebt een spelfout in een design token gecorrigeerd. Dit lijkt omdat het zo'n kleine wijziging is misschien een patch versie ophoging. Ook dit is feitelijk hetzelfde als het verwijderen van een design token gevolgd door het toevoegen van een nieuw design token. Dat maakt ook dit een breaking change.
```

> > Tip: Maak een verandern achterwaards compatibel

Reviewer 4: Zou mijn inziens ook de engels term mogen zijn als we het daarna kort toelichten? En anders zo...

```suggestion
> Tip: Maak een verandering achterwaarts compatibel.
```

> > In plaats van een design token in zijn geheel te verwijderen maak je een nieuw design te

Reviewer 4: Deze zin is nog niet helemal af? Voorzetje

```suggestion
> In plaats van een design token in zijn geheel te verwijderen maak je een nieuwe aan en verwijs je naar de oude.
```

> ### Tip, voorkom een breaking change en maak er een feature van

Reviewer 4: Volgens mij kan 'Tip' hier weg.
```suggestion
### Voorkom een breaking change en maak er een feature van
```

> In plaats van een design token van naam te veranderen `bijvoorbeeld button.background.colour: #fff` verwijderen maak je een nieuw design token aan `button.background.color: #fff` en je pas het oude design token aan naar het nieuwe: `button.background.colour: {button.background.color}`

Reviewer 4: De waarde van de tokens is niet nodig om dit punt te maken toch? Zo is de focus ook meer op de token naam (hetgeen dat wijzigt).

```suggestion
In plaats van een design token van naam te veranderen `bijvoorbeeld button.background.colour` verwijderen maak je een nieuw design token aan `button.background.color` en je pas het oude design token aan naar het nieuwe: `button.background.colour: {button.background.color}`
```

> Beide design tokens bestaan nu en je kunt aangeven dat mensen het nieuwe token wel moeten gebruiken en het oude token niet meer moeten gebruiken.

Reviewer 4: Vraag: Voor hoelang behoudt ik de 'depricated' token doen. Ik vol opruimwoede aankomen ;)

Wat normaal gesproken een breaking change zou zijn is hiermee een minor change geworden.

## Breaking changes in design tokens

### Wat zijn breaking changes?

> Breaking changes in design tokens zijn veranderingen die voor afnemers van je package negatieve gevolgen hebben. Het zijn veranderingen die niet achterwaarts compatibel zijn waardoor bestaande code niet meer of niet helemaal meer werkt zoals voorheen.

Reviewer 4: Package? Ik ben maar een simpele designer ;)

> Het resultaat hiervan is op de een of andere manier kapot. Soms is dit heel duidelijk, een achtergrondkleur die helemaal is verdwenen, soms is het subtieler en misschien niet meteen zichtbaar.

Reviewer 4: Klopt de zin zo ook wat jou betreft?

```suggestion
Hierdoor gaat er op een of andere manier iets kapot. Soms is dit heel duidelijk, een achtergrondkleur die helemaal is verdwenen, soms is het subtieler en misschien niet meteen zichtbaar.
```

> ### Voorbeelden van breaking changes in design tokens

Reviewer 4: Als we aan het begin van dit document toelichten dat dit document enkel over breaking changes m.b.t. design tokens gaan. Dan kan deze heading korter.

```suggestion
### Voorbeelden van breaking changes
```

> 

Reviewer 4: Zou het mooi zijn om deze sectie met voorbeelden subkopjes te geven? Bijvoorbeeld:
- Design token verwijderen
- Design token van naam veranderen
- Een spelfout corrigeren

> De eenvoudigste soort breaking change van een design token ontstaat als een design token helemaal wordt verwijderd. Gebruikers die het verwijderde design token in hun code gebruiken zien meteen dat hun code niet meer werkt zoals voorheen.

Reviewer 4: Idem. Als we bij een intro toelichten dat dit breaking changes m.b.t. design tokens betreft kunnen we dit soort zinnen korter maken.

```suggestion
De eenvoudigste soort breaking change ontstaat als een design token helemaal wordt verwijderd. Gebruikers die het verwijderde design token in hun code gebruiken zien meteen dat hun code niet meer werkt zoals voorheen.
```

> **Voorbeeld**: Het verwijderen van het design token `example.button.icon.space` zou ervoor zorgen dat de ruimte tussen het icoon en de tekst van een knop zou komen te vervallen. De button gaat door deze wijziging kapot.

Reviewer 4: ```suggestion
**Voorbeeld**: Het verwijderen van het design token `example.button.icon.space` zorgt ervoor dat de ruimte tussen het icoon en de tekst van een knop zou komen te vervallen. De button gaat door deze wijziging kapot.
```

> Een ander soort breaking change die hier erg op lijkt is het veranderen van de naam van een design token. Dat lijkt misschien een minder grote wijziging maar in feite komt het neer op:

Reviewer 4: ```suggestion
Een ander soort breaking change die hier op lijkt is het veranderen van de naam van een design token. Dat lijkt misschien een minder grote wijziging maar in feite komt het neer op:
```

> 1. het verwijderen van een design token(!)

Reviewer 4: ```suggestion
1. het verwijderen van een design token.
```

> 1. het toevoegen van een helemaal nieuw design token

Reviewer 4: ```suggestion
2. het toevoegen van een helemaal nieuw design token.
```

> **Voorbeeld**: Het veranderen van het design token `example.button.icon.space` naar `example.button.column-gap` zorgt ervoor dat de ruimte tussen een icoon en de tekst van een knop komt te vervallen. De button gaat dan ook door deze wijgiging kapot bij gebruikers van dit design token.

Reviewer 4: ```suggestion
**Voorbeeld**: Het veranderen van het design token `example.button.icon.space` naar `example.button.column-gap` zorgt ervoor dat de ruimte tussen een icoon en het label van een button komt te vervallen. De button gaat dan ook door deze wijgiging kapot bij gebruikers van dit design token.
```

Het aanpassen van maar 1 karakter in de naam van een design token, bijvoorbeeld het corrigeren van een typfout `colour` naar `color`, is al een breaking change.

> Breaking changes zijn echter, ondanks hun naam, niet per se erg. Het is wel heel belangrijk om er duidelijk over te communiceren. Meer daarover later.

Reviewer 4: ```suggestion
Breaking changes zijn, ondanks dat hun naam het doet vermoeden, niet per se erg. Het is echter wel heel belangrijk om er duidelijk over te communiceren. Daarover later meer.
```

## Plan je wijzigingen

Om te voorkomen dat een wijziging meteen een breaking change is zul je je wijziging goed moeten plannen.

Een strategie waar je voor kan kiezen is om meerdere design tokens naast elkaar te laten bestaan. Hierbij wordt een te vervangen token eerst gemarkeerd als achterhaald (in het Engels <span lang="en">“deprecated”</span>) en niet meteen helemaal verwijderd.

**Voorbeeld**: De naam van het design token `example.button.icon.space` moet veranderen in `example.button.column-gap`. In plaats van het oude design token meteen te verwijderen laat je de twee tokens naast elkaar bestaan.

> 1. maak eerst het nieuwe design token `example.button.column-gap` aan

Reviewer 4: Hier dan elke opsomming starten met een hoofdletter.

```suggestion
1. Maak eerst het nieuwe design token `example.button.column-gap` aan.
```

> 1. laat het oude design token `example.button.icon.space` verwijzen naar de waarde van het nieuwe design token, in Tokens Studio: `{example.button.column-gap}`

Reviewer 4: Tot nu toe is alles in Token Studio. Dus dat hoeft hier niet nog eens benoemd te worden.

```suggestion
2. Laat het oude design token `example.button.icon.space` verwijzen naar de waarde van het nieuwe design token `{example.button.column-gap}`.
```

> 1. markeer het oude design token `example.button.icon.space` als “deprecated” <!-- @todo: kan dit in Tokens Studio? -->

Reviewer 4: Ik zie 2 opties:
1. Depricated als 'Description' aan de token meegeven.
2. In Figma een extra 'sticker' toevoegen aan die balk zoals nu 'code-only' of 'figma-only'. Zie [Link](https://www.figma.com/design/shhwGcqPLi2CapK0P1zz8O/NLDS---Voorbeeld---Bibliotheek?node-id=462-1408&t=XvVvLsnqqbejH0O0-4) als voorbeeld.
```suggestion
3. Markeer het oude design token `example.button.icon.space` als 'deprecated' <!-- @todo: kan dit in Tokens Studio? -->
```

Door het oude token te behouden maar te markeren als “deprecated” vertel je gebruikers die het design token al gebruiken dat ze voor nieuwe code het design token niet meer moeten gebruiken en dat ze bestaande code zullen moeten aanpassen. Hoe code moet worden aangepast geef je aan in een changelog. Meer daarover hieronder.

## Communiceer over breaking changes

> Als je een [pull request hebt aangemaakt](/handboek/designer/stappenplan/#verstuur-aanpassingen-naar-github) met daarin breaking changes kun je daar in een extra commit informatie aan toevoegen om te aan te geven wat er veranderd is.

Reviewer 4: Klopt de zin zo nog steeds?

```suggestion
Als je een [pull request maakt](/handboek/designer/stappenplan/#verstuur-aanpassingen-naar-github) met breaking changes kun je daar informatie aan toevoegen waarin je aangeeft wat er veranderd is. Dit doe je doormiddel van een extra commit 
```

> ### Situatie A: in het NL Design System “themes” repository

Reviewer 4: ```suggestion
### Situatie A: in het NL Design System 'Themes' repository
```

Ga naar de lijst met [pull requests](https://github.com/nl-design-system/themes/pulls) en zoek de pull request met je wijzigingen op.

<!-- screenshot pull requests -->

> 1. Klik boven in de pagina op de lichtblauwe naam van de branch die je in Tokens Studio hebt opgegeven

Reviewer 4: ```suggestion
1. Klik boven in de pagina op de lichtblauwe naam van de branch die je hebt opgegeven.
```

> 1. Op de volgende pagina navigeer je naar de folder `.changesets`

Reviewer 4: ```suggestion
2. Op de volgende pagina navigeer je naar de folder `.changesets`
```

> 1. Kopieer het sjabloon uit het `README.md` bestand op deze pagina

Reviewer 4: ```suggestion
3. Kopieer het sjabloon uit het `README.md` bestand op deze pagina.
```

> 1. Klik rechtsboven op “Add a file” en kies voor ”Create new file”

Reviewer 4: ```suggestion
4. Klik rechtsboven op 'Add a file' en kies voor 'Create new file'.
```

> 1. Geef je bestand een naam bestaand uit drie willekeurige woorden en de extensie `.md`

Reviewer 4: ```suggestion
5. Geef je bestand een naam bestaand uit drie willekeurige woorden en de extensie `.md`.
```

> 1. Plak het gekopieerde sjabloon in de inhoud en pas de inhoud aan.

Reviewer 4: ```suggestion
6. Plak het gekopieerde sjabloon in de inhoud en pas de inhoud aan.
```

> 1. Klik rechtsboven op “Commit changes…”

Reviewer 4: ```suggestion
7. Klik rechtsboven op 'Commit changes…'
```

1. Geef een commit message op, bijvoorbeeld “Changeset toegevoegd”

> 1. Kies onderaan voor “Commit directly to the `<naam-van-je-branch>` branch

Reviewer 4: ```suggestion
9. Kies onderaan voor 'Commit directly to the `<naam-van-je-branch>` branch'.
```

> 1. Klik op “Commit changes”

Reviewer 4: ```suggestion
10. Klik op 'Commit changes'.
```

### Situatie B: in je eigen repository

> <!-- @todo willen we dit onderscheid? -->

Reviewer 4: Vragen die ik na het lezen zelf nog heb:

- Wat nu als de design token gaat plaats maken voor een vaste waarde? button.border-width krijgt in css een fixed value (border-width: 1px) want is bijvoorbeeld een best practise en blijkt (bij candiate) voor elke org zo prima te werken?
- Wat nu als dit ook aan de code kant wordt doorgevoerd? Nog steeds breaking?
- Wat nu als er juist een breaking change aan de code kant wordt doorgevoerd? Hoe weet ik dat als designer? Keertje proefdraaien?
